tests/template_tests/test_logging.py

```diff
@@ -75,7 +75,22 @@ def test_log_on_variable_does_not_exist_not_silent(self):
         raised_exception = self.test_handler.log_record.exc_info[1]
         self.assertEqual(
             str(raised_exception),
-            'Failed lookup for key [author] in %r' % ("{%r: %r}" % ('section', 'News'))
+            'Failed lookup for key [author] in %r' % ({'section': 'News'},)
+        )
+
+    def test_log_on_variable_does_not_exist_does_not_stringify(self):
+        class TestObject:
+            def __str__(self):
+                return 'fail'
+
+        test_object = TestObject()
+
+        with self.assertRaises(VariableDoesNotExist) as raised_exception:
```

Comment:
I was thinking you could instantiate `VariableDoesNotExist` directly rather than creating it indirectly with `Variable/resolve()`. Not sure about a better place for this test considering there's no logging involved at this point. Maybe `tests.py` is for miscellaneous tests.